core/src/erasure.rs

```diff
@@ -815,29 +656,32 @@ pub mod test {
             let n = *i as usize;
             let mut idx = n as u64 + block_start_idx;
 
-            let mut data_size;
+            let data_size;
             if n < NUM_DATA {
-                data_size = locks[n].data_size() as usize;
-                data_size -= BLOB_HEADER_SIZE;
-                if data_size > BLOB_DATA_SIZE {
+                data_size = locks[n].data_size() as usize - BLOB_HEADER_SIZE;
+
+                if data_size > BLOB_DATA_SIZE + BLOB_HEADER_SIZE {
```

Comment:
this looks like you've subtracted BLOB_HEADER_SIZE from the calculation twice...  was it incorrect before?  can you double-check this math, write a test that hits this clause?

Reply:
Yes I think this is wrong. I'll add a test.

Hopefully I can change blobs soon and this kind of stuff can go away.